src/libs/AppStateMonitor/index.js

```diff
@@ -0,0 +1,35 @@
+import {AppState} from 'react-native';
+import CONST from '../../CONST';
+import shouldReportActivity from './shouldReportActivity';
+
+let appState = CONST.APP_STATE.ACTIVE;
+
+/**
+ * Listener that will only fire the callback when the user has become active.
+ *
+ * @param {Function} callback
+ * @returns {Function} to unsubscribe
+ */
+function addBecameActiveListener(callback) {
+    /**
+     * @param {String} state
+     */
+    function appStateChangeCallback(state) {
+        if (
+            shouldReportActivity
+            && (appState === CONST.APP_STATE.INACTIVE || appState === CONST.APP_STATE.BACKGROUND)
+            && state === CONST.APP_STATE.ACTIVE
+        ) {
+            callback();
+        }
+        appState = state;
+    }
+    AppState.addEventListener('change', appStateChangeCallback);
+    return () => {
+        AppState.removeEventListener('change', appStateChangeCallback);
+    };
```

Comment:
Could you explain what is happening here? I'm not totally sure what this is doing, is it initializing with an event listener and then whenever it is returned it gets removed?

Reply:
Sure! So, `addBecameActiveListener` is both setting up an event listener and then returning a function that we can call later to remove the listener. The `appStateChangeCallback` exists in the closure created by calling the outermost function. It's just sort of a convenient pattern for cleaning up the listener we created and I borrowed it from how `NetInfo` works.